src/backend/distributed/commands/foreign_constraint.c

```diff
@@ -741,31 +756,71 @@ TableReferencing(Oid relationId)
 
 
 /*
- * ConstraintIsAForeignKey is a wrapper around GetForeignKeyOidByName that
- * returns true if the given constraint name identifies a foreign key
- * constraint defined on relation with relationId.
+ * ConstraintWithNameIsOfType is a wrapper around ConstraintWithNameIsOfType that returns true
+ * if given constraint name identifies a uniqueness constraint, i.e:
+ *   - primary key constraint, or
+ *   - unique constraint
+ */
+bool
+ConstraintIsAUniquenessConstraint(char *inputConstaintName, Oid relationId)
+{
+	/* TODO: should we consider exclusion as well ? */
```

Comment:
A question for me to research: Can foreign keys be defined on exclusion constraints ?

Reply:
Obviously no :)